packages/modern-test-app/app/components/helpers.hbs

```diff
@@ -12,6 +12,6 @@
 {{xor true false}}
 
 <AndOrTypeChecking
-  @andArg={{or (hash) (array)}}
-  @orArg={{and (hash) (array)}}
+  @andArg={{or this.objectOrUndefined this.arrayOrUndefined}}
+  @orArg={{and this.objectOrUndefined this.arrayOrUndefined}}
```

Comment:
```hbs
@andArg={{or ...
@orArg={{and ...
```

I don't quite understand the test but, are these the wrong way around?

Reply:
yeah, this test case needs updates to better reflect use real world scenarios.
Will do the same prob after #196